### PR TITLE
Upload documentation PRs to Netlify

### PR DESCRIPTION
**Pull request by t3chguy**
Signed-off-by: Michael Telatynski <7t3chguy@gmail.com>

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

**Comment by DMRobertson** [comment]
@matrix-org/synapse-core are we interested in this?

**Comment by t3chguy** [comment]
I was told by @babolivier that it was desired, then later @erikjohnston was looking into it when I pointed him at this PR

**Comment by erikjohnston** [comment]
> I was told by @babolivier that it was desired, then later @erikjohnston was looking into it when I pointed him at this PR

Its taken me a while to remember why I was! The answer is that it would be useful to have the rust docs uploaded for things that touch the rust code

File: .github/workflows/docs-pr-netlify.yaml
Comment (babolivier) [inline .github/workflows/docs-pr-netlify.yaml:17]: Why are we using commit refs instead of tags?

File: .github/workflows/docs-pr-netlify.yaml
Comment (babolivier) [inline .github/workflows/docs-pr-netlify.yaml:32]: Are these org- or repo-level secrets? If the latter it would be good to make sure the secrets are set before we merge this.

**Review by babolivier — CHANGES_REQUESTED** [review]
Just a couple of questions.

File: .github/workflows/docs-pr-netlify.yaml
Comment (t3chguy) [reply .github/workflows/docs-pr-netlify.yaml:17]: Tags can be replaced, breaking a hash is far more difficult

File: .github/workflows/docs-pr-netlify.yaml
Comment (t3chguy) [reply .github/workflows/docs-pr-netlify.yaml:32]: the token could be org-level, but the site id needs to be repo level to match the project

**Comment by t3chguy** [comment]
> Also has this been tested e.g. on a branch?

No due to lack of access to add secrets to this repo, but it is basically a clone of how Element Web's Netlify works so it should work first time every time ™️

File: .github/workflows/docs-pr-netlify.yaml
Comment (babolivier) [reply .github/workflows/docs-pr-netlify.yaml:17]: Is this a thing we're afraid can realistically happen? I'm unconvinced maintainers of the two third-party workflows we use would ever do this, and even then it doesn't seem very difficult to fix to me.

Comment (t3chguy) [reply .github/workflows/docs-pr-netlify.yaml:17]: FTR it is copied from matrix-spec, so it'd be good to be org-consistent: https://github.com/matrix-org/matrix-spec/blob/main/.github/workflows/netlify.yaml#L37-L42

File: .github/workflows/docs-pr-netlify.yaml
Comment (babolivier) [reply .github/workflows/docs-pr-netlify.yaml:32]: Token has been set by Rich on the repo.

**Comment by babolivier** [comment]
From DMs with @t3chguy: this is relying on catching a `workflow_run` event from one of the new workflows, which only works if the workflow lives on the default branch. So I'm going to merge this PR, test that it works on a draft PR, and revert if this fails.

**Review by babolivier — APPROVED** [review]

**Comment by t3chguy** [comment]
For context this is using `workflow_run` to be able to work on PRs from forks, using `pull_request` would not have access to the Netlify secrets in those PRs.

**Pull request merged**
